core/sr-primitives/src/generic/tip.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Tip structure for a transaction.
+
+use crate::codec::{Encode, Decode};
+use crate::traits::Zero;
+
+/// A placeholder tip type that is used to fulfill the generic requirements of `checked_extrinsic`
+/// when the underlying `unchecked_extrinsic` actually does not have a tip.
+pub type NoTipBalance = u32;
```

Comment:
why not `()`?

Reply:
Could not have it meet the type requirements of the generic type (has to be `SimpleArithmetic`-- probably doable but needs a more handy rust dev with more time -- will try)

Reply:
For future reference this is enough: 

```
impl<Balance: Zero> From<()> for Tip<Balance> {
    fn from(_: ()) -> Self {
        Tip::Sender(Zero::zero())
    }
}
```

No need to have `default` actually.